universum/modules/vcs/vcs.py

```diff
@@ -172,10 +172,12 @@ def clean_sources_silently(self):
         except OSError:
             pass
 
-    def supports_copy_cl_files_and_revert(self):
-        return self.driver.supports_copy_cl_files_and_revert
+    def copy_cl_files_and_revert(self) -> None:
+        raise NotImplementedError("Not applicable")
 
     @make_block("Revert repository")
-    def revert_repository(self):
-        diff = self.driver.copy_cl_files_and_revert()
-        return diff
+    def revert_repository(self) -> Optional[List[Tuple[Optional[str], Optional[str], Optional[str]]]]:
+        if isinstance(self.driver, base_vcs.BaseDownloadVcs):
```

Comment:
Is this pythonic?

Reply:
nope, but it calms mypy (=
I am still thinking about a good way to resolve this. I can do try catch and re-raise instead (like in 'main'), but then mypy would might want me to move 'copy_cl_files_and_revert' to common base class ('BaseDownloadVcs')

Reply:
the problem is if this gets called when driver is not BaseDownloadVcs

Reply:
for now I moved 'copy_cl_files_and_revert' to BaseVcs to resolve this. It's always available, but in most cases causes NotImplementedError